src/govuk/settings/_colours-palette.scss

```diff
@@ -26,6 +28,16 @@ $govuk-use-legacy-palette: if(
   false
 ) !default;
 
+// Only show the deprecation warning if user is setting $govuk-use-legacy-palette
+// manually instead of automatically via compatibility variables
+@if $govuk-use-legacy-palette == true and
+  $govuk-compatibility-govukfrontendtoolkit == false and
+  $govuk-compatibility-govuktemplate == false and
+  $govuk-compatibility-govukelements == false {
```

Comment:
If this was moved above the definition I think you could just do…

```scss
@if variable-exists($govuk-use-legacy-palette) {
```

As the only way the variable would exist at that point is if it was set in userspace.

(Same applies for the other settings)

Reply:
Sadly I haven't been able to get this to work. It seems that `variable-exists` and `global-variable-exists` both scope existing variables globally and doesn't cascade, meaning that even if I move this above the definition, it still exists in the sass as a whole so it's still true locally. This also applies if I add `and $govuk-use-legacy-palette == true` as this seems to be informed by the compatibility mode variables still, meaning that I still have to check that the compatibility mode variables are false or get bombarded by warnings when running locally. I don't know if I've done something wrong here or if `variable-exists` is too loose for this use case.